Tests/Services/HealthTests/HealthTests.swift

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import GRPCHealth
+import GRPCInProcessTransport
+import XCTest
+
+@available(macOS 15.0, iOS 18.0, watchOS 11.0, tvOS 18.0, visionOS 2.0, *)
+final class HealthTests: XCTestCase {
+  private func withHealthClient(
+    _ body: @Sendable (Grpc_Health_V1_HealthClient, HealthProvider) async throws -> Void
+  ) async throws {
+    let health = Health()
+    let inProcess = InProcessTransport.makePair()
+    let server = GRPCServer(transport: inProcess.server, services: [health.service])
+    let client = GRPCClient(transport: inProcess.client)
+    let healthClient = Grpc_Health_V1_HealthClient(client: client)
+
+    try await withThrowingDiscardingTaskGroup { group in
+      group.addTask {
+        try await server.run()
+      }
+
+      group.addTask {
+        try await client.run()
+      }
+
+      do {
+        try await body(healthClient, health.provider)
+      } catch {
+        XCTFail("Unexpected error: \(error)")
+      }
+
+      group.cancelAll()
+    }
+  }
+
+  func testCheckOnKnownService() async throws {
+    try await withHealthClient { (healthClient, healthProvider) in
+      let testServiceDescriptor = ServiceDescriptor(package: "test.package", service: "TestService")
```

Comment:
Rather than spelling out the descriptor every time we can add an extension within this file to `ServiceDescriptor`, e.g.:

```swift
extension ServiceDescriptor {
  fileprivate static let echo = ServiceDescriptor(package: "echo", service: "Echo")
}
```